### PR TITLE
fix bug for models with mask=False

### PR DESCRIPTION
**Pull request by fehiepsi**
Addresses #568.

### CODE REVIEW

File: numpyro/infer/util.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def log_density(model, model_args, model_kwargs, params, skip_dist_transforms=Fa
     else:
         model = substitute(model, param_map=params)
     model_trace = trace(model).get_trace(*model_args, **model_kwargs)
-    log_joint = 0.
+    log_joint = jax.device_put(0.)
```
Comment (neerajprad) [inline numpyro/infer/util.py:62]: @fehiepsi - maybe we should also add a check to `mask` so that it raises a ValueError if fn is not a callable to ensure that users don't trip up by not specifying this as a keyword:

```python
if fn is not None and not callable(fn):
    raise ValueError("Expected `fn` to be callable; instead found type(fn) = {}.".format(type(fn)))
```

File: numpyro/infer/util.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def log_density(model, model_args, model_kwargs, params, skip_dist_transforms=Fa
             scale = site['scale']
             # Early exit when all elements are masked
             if not_jax_tracer(mask) and mask is not None and not np.any(mask):
-                return jax.device_put(0.), model_trace
+                continue
```
Comment (neerajprad) [inline numpyro/infer/util.py:71]: Thanks for fixing this. So if everything is masked this will return `0` as the log density?

File: numpyro/infer/util.py (hunk 1 above)
Comment (fehiepsi) [reply numpyro/infer/util.py:62]: good point! I'll add it.

File: numpyro/infer/util.py (hunk 2 above)
Comment (fehiepsi) [reply numpyro/infer/util.py:71]: Yes. We start looping over the sites with `log_joint = 0.`

File: numpyro/primitives.py [hunk 3]
```diff
@@ -42,6 +42,9 @@ def apply_stack(msg):
 
 class Messenger(object):
     def __init__(self, fn=None):
+        if fn is not None and not callable(fn):
+            raise ValueError("Expected `fn` to be callable; instead found type(fn) = {}."
```
Comment (neerajprad) [inline numpyro/primitives.py:46]: nit: can we change `to be callable` --> `to be a Python callable object` for clarity?

Comment (fehiepsi) [reply numpyro/primitives.py]: thanks
